File: backend/app/api/v1/schemas/health.py
from datetime import datetime
from typing import Optional, Dict, Any
from pydantic import BaseModel


class ServiceComponentHealth(BaseModel):
    status: str
    latency_ms: Optional[float] = None
    error: Optional[str] = None


class HealthResponse(BaseModel):
    status: str
    app: str
    environment: str
    version: str
    timestamp: datetime
    services: Dict[str, ServiceComponentHealth]

File: frontend/src/api/health.ts
const API_BASE = import.meta.env.VITE_API_URL || '/api/v1';

export interface HealthResponse {
  status: 'HEALTHY' | 'DEGRADED' | string;
  app: string;
  environment: string;
  version: string;
  timestamp: string;
  services: {
    database: { status: string; latency_ms?: number };
    redis: { status: string; latency_ms?: number };
  };
}

export async function getSystemHealth(): Promise<HealthResponse> {
  const res = await fetch(`${API_BASE}/health`);
  if (!res.ok) {
    throw new Error(`Health check failed: ${res.statusText}`);
  }
  return res.json();
}
